[draft] PP with compile test

H-Huang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #143386
* #143599

testing command:
`python test/distributed/_composable/test_composability/test_pp_composability.py -k test_pp_with_compile`

cc @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

H-Huang committed:
[draft] PP with compile test

[ghstack-poisoned]

wconstab commented:
we may want a one-off test (rather than parametrize) where we just ensure that pp+zb+compile raises an expected runtime error rather than segfaulting

H-Huang committed:
Update on "[draft] PP with compile test"


testing command:
`python test/distributed/_composable/test_composability/test_pp_composability.py -k test_pp_with_compile`

cc awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o

[ghstack-poisoned]

H-Huang committed:
Update on "[draft] PP with compile test"


testing command:
`python test/distributed/_composable/test_composability/test_pp_composability.py -k test_pp_with_compile`

cc awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o

[ghstack-poisoned]
